specs/BadgeNotifications/BadgeNotifications-spec.md

```diff
@@ -0,0 +1,192 @@
+| Unavailable | A glyph that shows the user or a service is not currently available or offline. | ![Screenshot](badge-unavailable.png) |
+| Error | A glyph that denotes an error has occurred, which may require user action to resolve. | ![Screenshot](badge-error.png) |
+
+These glyphs are system-provided and standardized, ensuring a consistent look and feel across different applications that use them. By using these glyphs, developers can convey specific states or notifications to users in a visually intuitive manner.
+
+# API Details
+
+> Note: all of this new WinAppSDK API is to support 
+  Badge Notifications in WindowsAppSdk
+ 
+```c++ (but really MIDL3)
+namespace Microsoft.Windows.BadgeNotifications
+{
+    // Set of predefined glyphs that can be used to represent various statuses or notifications on an application's badge
+    enum BadgeGlyphValue
+    {
+        None, // No glyph. A blank tile appears in the badge.
+        Activity, // A glyph representing an activity
+        Alert, // A glyph representing an alert
+        Alarm, // A glyph representing an alarm
+        Attention, // A glyph representing attention
+        Available, // A glyph representing availability
+        Away, // A glyph representing being away
+        Busy, // A glyph representing being busy
+        NewMessage, // A glyph representing a new message
+        Paused, // A glyph representing being paused
+        Playing, // A glyph representing playing
+        Unavailable, // A glyph representing being unavailable
+        Error, // A glyph representing an error
+    };
+
+    runtimeclass BadgeNotification
+    {
+        // Constructs a BadgeNotification with a numeric badge value.
+        // This constructor initializes the BadgeNotification with a number that will be displayed on the badge.
+        // @param num The numeric value to be displayed on the badge.
+        BadgeNotification(UInt32 num);
+
+        // Constructs a BadgeNotification with a glyph value.
+        // This constructor initializes the BadgeNotification with a predefined glyph from the BadgeGlyphValue enum.
+        // @param value The BadgeGlyphValue representing a system-provided glyph to be displayed on the badge.
+        BadgeNotification(BadgeGlyphValue value);
+
+        // Gets or sets the time after which a Badge Notification should not be displayed.
+        Windows.Foundation.DateTime Expiration;
+    }
+
+    // The manager class which encompasses all Badge Notification API Functionality
+    runtimeclass BadgeNotificationManager
+    {
+        // Gets a Default instance of a BadgeNotificationManager
+        static BadgeNotificationManager Default{ get; };
+
+        // Applies a change to the badge's glyph or number.
+        void UpdateBadge(BadgeNotification notification);
+
+        // Removes the Badge Notifications for the App from Action Centre
+        Windows.Foundation.IAsyncAction ClearBadge();
```

Comment:
Why is "Clear" an async operation but Update is synchronous?

Reply:
The `Update` method is synchronous because displaying a notification is usually a quick operation that does not require significant processing or waiting for external resources. When an app calls `Update`, it typically expects the notification to appear immediately or nearly immediately, and there is no need to wait for a potentially long-running operation to complete. Therefore, making this method synchronous allows for a simple and straightforward usage pattern.

On the other hand, the `Clear` method is asynchronous because it involves more than just a simple display action. Removing a notification could potentially require interaction with the system's notification service, which involve I/O operations, updating the notification database, and ensuring that the removal is properly reflected in the Action Center. These operations can take more time and are subject to delays that are outside the control of the application.

By making `Clear` asynchronous, the Windows App SDK allows the calling thread to continue executing while the notification removal process is underway. This prevents the app from becoming unresponsive if the removal operation takes longer than expected. The async pattern also provides a way to handle the completion of the operation with a callback or continuation, which is useful for updating the app's state or UI after the notification has been successfully removed.

In summary, the decision to make `Update` synchronous and `Clear` asynchronous is based on the expected complexity and duration of the operations. 

Similar logic has been used for AppNotification in WindowsAppSdk - https://github.com/microsoft/WindowsAppSDK/blob/b0f4e3477ec6be65f674aee4624f993e1845560a/specs/AppNotifications/AppNotifications-spec.md

Reply:
ClearBadge has been updated to Synchronous